keel-core/src/main/kotlin/com/netflix/spinnaker/keel/constraints/ConstraintEvaluator.kt

```diff
@@ -31,6 +33,9 @@ interface ConstraintEvaluator<T : Constraint> {
     artifact: DeliveryArtifact,
     version: String,
     deliveryConfig: DeliveryConfig,
-    targetEnvironment: String
+    targetEnvironment: Environment,
+    priorState: ConstraintState? = null
   ): Boolean
 }
+
+interface StatefulConstraintEvaluator<T : Constraint> : ConstraintEvaluator<T>
```

Comment:
Does this really need to be a separate interface?

Reply:
Nope, will remove. I made it so that I can filter evaluators on stateful vs. stateless to enforce evaluation order, but looks like I can do that fine via the original interface via: 
``` 
private val statefulEvaluators: List<ConstraintEvaluator<*>> = constraints
    .filter { it.constraintType is StatefulConstraint}
```